### PR TITLE
Add optional feature flag that creates empty session if session is not found

### PR DESCRIPTION
**Pull request by geek-soft**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Note: Cookie parsing must happen before request is passed to session middleware.
 Since build artifacts are stored in the repository (no NPM registry in used just yet) to bring this module as dependency please add the following fragment to `package.json`: 
 
 ```$json
-"ch-node-session-handler": "git+ssh://git@github.com/companieshouse/node-session-handler.git#3.0.5"
+"ch-node-session-handler": "git+ssh://git@github.com/companieshouse/node-session-handler.git#3.1.0"
 ```
 
 ### Session
```
Comment (akaJSaunders) [inline README.md:33]: I think its worth adding the new option on the README.md

File: README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Note: Cookie parsing must happen before request is passed to session middleware.
 Since build artifacts are stored in the repository (no NPM registry in used just yet) to bring this module as dependency please add the following fragment to `package.json`: 
 
 ```$json
-"ch-node-session-handler": "git+ssh://git@github.com/companieshouse/node-session-handler.git#3.0.5"
```
Comment (akaJSaunders) [inline README.md:30]: Could you explain how the version bumping works? The option has a value of false by default, so it shouldn't break any existing implementation right?

Comment (geek-soft) [reply README.md:30]: Semantic versioning has a 3 digits form `x.y.z`. Last one (z) is incremented when software is patched for example bug fixes. Middle one (y) is changed when new feature is added. Here support for session without sign in is considered a new feature. Finally first one (x) is incremented when breaking changes are introduced. Here API is backward compatible so I decided to increment middle ingredient only.

Comment (geek-soft) [reply README.md:30]: On a second thought I might consider bumping first digit in response to validation change.

Comment (geek-soft) [reply README.md:30]: I believe validation change is safe cause node security modules verifies whether sign in info exist but as a precaution I will bump version to 4.0.0

File: README.md (hunk 1 above)
Comment (geek-soft) [reply README.md]: Added

File: README.md (hunk 2 above)
Comment (akaJSaunders) [reply README.md:30]: Thanks for the explanation!

**Review by akaJSaunders — APPROVED** [review]

**Pull request merged**